trio/_core/_run.py

```diff
@@ -1376,7 +1377,13 @@ def run_impl(runner, async_fn, args):
             except StopIteration as stop_iteration:
                 final_result = Value(stop_iteration.value)
             except BaseException as task_exc:
-                final_result = Error(task_exc)
+                # Store for later, removing uninteresting top frames:
+                #   1. task.context.run()
+                #   2. contextvars.callable() (< Python 3.7 only)
```

Comment:
Huh, I thought the top two frames were `trio._core._run.run_impl` and (on <3.7 only) `contextvars.Context.run`. Do you know where the discrepancy comes from? am I confused? :-)

Reply:
here's what you wrote in #56:

```
  File "/.../site_packages/trio/_core/_run.py", line 1374, in run_impl
    msg = task.context.run(task.coro.send, next_send)
```
> This is noise... when we catch a user exception in run_impl, we should unconditionally discard the top frame in the traceback (which will be this one).

```
  File "/.../site-packages/contextvars/__init__.py", line 38, in run
    return callable(*args, **kwargs)
```
> This line will disappear on 3.7, where contextvars.Context.run is implemented in C, so maybe not worth worrying about. Or we could discard it at the same time we fix the run_impl frame above.

Reply:
I see, I'm enumerating the child functions rather than the parents.

Reply:
Ahhhh yeah, right, so we're just looking at different lines in the traceback :-). that makes sense!